src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,360 @@
+import { Action, Getter } from 'vuex-class'
+
+@Component({
+  components: {
+    AddNameTranslation,
+    ListNameTranslation,
+    ConfirmDialog
+  }
+})
+export default class NameTranslation extends Mixins(CommonMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationDialog: ConfirmDialogType
+  }
+
+  @Prop({ default: () => [] })
+  private nameTranslations!: NameTranslationDraftIF[]
+
+  private actionTypes = ActionTypes
+
+  // Properties
+  private draftTranslations: NameTranslationDraftIF[] = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasPendingChange (): boolean {
+    return this.draftTranslations.length !== this.nameTranslations.length ||
+      !this.draftTranslations.every((translation, index) => {
+        return this.nameTranslations[index].value === translation.value
+      })
+  }
+
+  private get hasNameTranslationChange (): boolean {
+    return this.draftTranslations.length > 0 &&
+      this.draftTranslations.filter(x => x.action).length > 0
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.draftTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.draftTranslations = this.nameTranslations
+      .filter(x => x.action !== 'added')
+      .map(a => {
+        const translation = Object.assign({}, a)
+        translation.value = translation.oldValue || translation.value
+        translation.oldValue = null
+        translation.action = null
+        return translation
+      })
+    this.emitNameTranslations(this.draftTranslations)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private cancelNameTranslationCorrection () {
+    const nameTranslations = this.nameTranslations || []
+    const hasUnsavedData = this.draftTranslations.length !== nameTranslations.length ||
+      !this.draftTranslations.every((translation, index) => {
+        return nameTranslations[index].value === translation.value &&
+          nameTranslations[index].action === translation.action
+      })
+    if (hasUnsavedData) {
+      this.confirmUnsavedChanges()
+    } else {
+      this.isEditing = false
+    }
+  }
+
+  private confirmUnsavedChanges () {
+    // open confirmation dialog and wait for response
+    this.$refs.confirmTranslationDialog.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Name Translations. Do you want to save these changes?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Save changes',
+        no: null,
+        cancel: 'Cancel without saving'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.setNameTranslations()
+      }
+    }).catch(() => {
+      this.draftTranslations = this.nameTranslations ? this.nameTranslations.map(a => Object.assign({}, a)) : []
```

Comment:
Yep, good catch -- your previous `[...x]` code only did a shallow copy of the array. THIS code returns an array with new items (but remember that assign() also does a shallow copy so you have to watch that your items don't have sub-objects).
FWIW, an alternative is Lodash cloneDeep(). Up to you.